### PR TITLE
Reimplement embulk-guess-msgpack in Java

### PR DESCRIPTION
**Pull request by dmikurube**
To reduce dependencies on JRuby in plugins under https://github.com/embulk, reimplementing the guess plugin for MessagePack in Java.

### CODE REVIEW

File: embulk-guess-msgpack/src/main/java/org/embulk/guess/msgpack/MsgpackGuessPlugin.java [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+            } else if (fileEncoding.equals("sequence")) {
+                // do nothing
+            }
+
+            final List<ConfigDiff> schema;
+
+            if (rowEncoding.equals("map")) {
+                final ArrayList<LinkedHashMap<String, Object>> rows = new ArrayList<>();
+                try {
+                    while (true) {
```
Comment (hiroyuki-sato) [inline embulk-guess-msgpack/src/main/java/org/embulk/guess/msgpack/MsgpackGuessPlugin.java:97]: It seems that `IllegalStateException` isn't a subclass of the `IOException`. 

Is this part completely the same as the original?

```ruby
        begin
          while true
            rows << JSON.parse(uk.unpackValue.toJson)
          end
        rescue java.io.EOFException
        end
```

**Review by hiroyuki-sato — COMMENTED** [review]
Could you check my comment just in case?

File: embulk-guess-msgpack/src/main/java/org/embulk/guess/msgpack/MsgpackGuessPlugin.java (hunk 1 above)
Comment (hiroyuki-sato) [reply embulk-guess-msgpack/src/main/java/org/embulk/guess/msgpack/MsgpackGuessPlugin.java:97]: Oh, I understand this part after I commented, that the parser raises `IOException` when parsing reaches the end of the part. So, This is completely the same original part.

**Review by hiroyuki-sato — APPROVED** [review]
LGTM👍

File: embulk-guess-msgpack/src/main/java/org/embulk/guess/msgpack/MsgpackGuessPlugin.java (hunk 1 above)
Comment (dmikurube) [reply embulk-guess-msgpack/src/main/java/org/embulk/guess/msgpack/MsgpackGuessPlugin.java:97]: Yeah, this `IOException` seems like an "expected" Exception -- it's a bad pattern, but unavoidable for such a guessing logic.

**Comment by dmikurube** [comment]
Thanks!

**Pull request merged**